### PR TITLE
feat: Added banner for warning on old version

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

A lot of support requests simply come down to people having a really old version of the JS.

Solves https://github.com/PostHog/posthog/issues/15781

## Changes

* Adds a banner (In recordings only for now) that informs the user their version is behind.
* Dismissible by the current version / available version pair
* Only shows if the diff is greater than 5, allowing us to have the time difference of deploying out new versions
* Shows it as a warning/error after a certain threshold diff
* Checks periodically, persisting the value so that we don't over fetch

Screenshot 2023-07-31 at 10 12 01
Screenshot 2023-07-31 at 10 12 13
Screenshot 2023-07-31 at 10 11 35

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/mocks/handlers.ts [hunk 1]
```diff
@@ -80,6 +80,8 @@ export const defaultMocks: Mocks = {
             error: 'Error fetching stats from sentry',
             exception: "[ErrorDetail(string='Sentry integration not configured', code='invalid')]",
         },
+        // We don't want to show the "new version available" banner in tests
+        'https://api.github.com/repos/posthog/posthog-js/tags': () => [200, []],
```
Comment (pauldambra) [inline frontend/src/mocks/handlers.ts:84]: makes me wonder if we should then have a story that specifically shows it but 🤷

**Review by pauldambra — APPROVED** [review]
👍 a couple of thoughts, but this comes up so much!

File: frontend/src/lib/components/VersionChecker/versionCheckerLogic.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                },
+            },
+        ],
+        usedVersions: [
+            null as SDKVersion[] | null,
+            {
+                loadUsedVersions: async () => {
+                    const query: HogQLQuery = {
+                        kind: NodeKind.HogQLQuery,
+                        query: `SELECT properties.$lib_version AS lib_version, max(timestamp) AS latest_timestamp, count(lib_version) as count
```
[the remark this answers is not in the document]
Comment (benjackwhite) [reply frontend/src/lib/components/VersionChecker/versionCheckerLogic.ts:44]: Generally I like it but I think that should be a follow up.

File: frontend/src/lib/components/VersionChecker/versionCheckerLogic.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            }
+
+            let diff = values.availableVersions.findIndex((v) => v.version === currentVersion)
+            diff = diff === -1 ? values.availableVersions.length : diff
+
+            const warning: SDKVersionWarning = {
+                currentVersion,
+                latestVersion,
+                diff,
+                level: diff > 20 ? 'error' : diff > 10 ? 'warning' : 'info',
```
Comment (daibhin) [inline frontend/src/lib/components/VersionChecker/versionCheckerLogic.ts:105]: Didn't think about it until the demo just now but I wonder if a better metric for severity would be patch, minor and major updates? Might not be super easy to calculate but just a thought

**Pull request merged**